libservice/headers/service/Service.h

```diff
@@ -16,11 +16,26 @@
 
 #pragma once
 
+#include <chrono>
 #include <cstdint>
+#include <iostream>
 #include <string>
+#include <unordered_map>
+#include <vector>
 
 namespace service {
 
+struct ArrayHasher {
+	template<size_t N>
+	std::size_t operator()(const std::array<uint8_t, N>& array) const {
```

Comment:
Are ipv4 or ipv6 address not unique? Maybe identity function can be used as a hash?  Do you really need this inefficient hash function which iterates over bytes?